plugins/Observability/src/test/report-generator-for-regression-testing.ssc

```diff
@@ -0,0 +1,63 @@
+// Name: Observability Plugin Report Generator For Regression Testing
+// License: Public Domain
+// Author: colossatr0n
+// Version: 1.0
+// Description: Generates reports to be used for regression testing. Run this script in two different versions of Stellarium and compare the output using some kind of differ (vim, git, diff). For macOS, file is output to "~/Library/Application Support/Stellarium/".
+
+const objectNames = [
+    "Sun",      // Sun
+    "Moon",     // Moon
+    "Jupiter",  // Solar system planet
+    "Achernar", // Interstellar object
+]
+
+// Locations from different continents
+const observerLocations = [
+    "Boise, Northern America",
+    "Brasilia, Southern America",
+    "Palestine, Northern America",
+    "Barnaul, Northern Asia",
+    "Albany, Australasia",
+    "Prague, Eastern Europe"
+]
+
+Observability.showReport(true)
+
+core.resetOutput()
+var reports = []
+for (i=0; i < observerLocations.length; i++) {
+    var location = observerLocations[i]
+    core.setObserverLocation(location, 0.0)
+    var actualLocation = core.getObserverLocation()
+    var attempts = 0
+    while (location != actualLocation) {
+        if (attempts > 100) {
+            core.debug("Couldn't update location. It appears that the location ID '" + location + "' is incorrect or the attempted location change happened too slowly.")
+            break
+        }
+        actualLocation = core.getObserverLocation()
+        attempts++
+    }
+    for (j=0; j < objectNames.length; j++) {
+        var objectName = objectNames[j]
+        core.selectObjectByName(objectName, false)
+        // Need to wait a sufficient amount of time to allow the object selection to take place.
+        // Could use core.getSelectedObjectInfo to verify the selection was successful, but using that method crashes Stellarium.
+        core.wait(0.1)
+
+        var reportJson = Observability.getReportAsJson()
+        var reportObj = JSON.parse(reportJson)
+
+        var augmentedReportObj = { objectName: objectName, location: actualLocation, report: reportObj }
+        reports.push(augmentedReportObj)
+    }
+}
+core.output(JSON.stringify(reports))
+
+const d = new Date()
+time = d.getTime()
+// For macOS, output will be saved to ~/Library/Application Support/Stellarium/observability-report.json
+core.saveOutputAs("observability-report-" + time + ".json")
+core.debug("For macOS, output was saved to:")
```

Comment:
This should be inhibited on non-Macs. Actually it is not needed, this is just the default user data directory of which the user should be aware of.

Reply:
Just pushed an update to make it platform specific. 

The output is still useful and makes it easy to copy and paste the path into a terminal, especially given that the filename has a timestamp appended to the end.

I've been a macOS user for a long time and didn't know that `~/Library/Application Support/` would be considered the "user data" directory. I've always called it an "application data" directory. I was looking under `/Users/username` at first because I considered "user data" to be synonymous with the home directory. I imagine other macOS users might make the same assumptions.

Reply:
I don't oppose the timestamp. Just that Stellarium tries to be cross-platform, so Mac-specific notes should be only given where it is in any way exclusive or special. But this same script should run on Linux and Windows, writing to the user data directory as expected and described in the User Guide, chapter 5. So, please just remove lines 59, 61&62. Or write

core.debug("Output was saved to: observability-report-" + time + ".json")

Reply:
@gzotti Can you review the updated version of this file?

Reply:
Oh wait, maybe I didn't push..checking.

Reply:
Pushed.